Intro_to_Base_R: Switch iris -> penguins dataset (#403)
AlexsLemonade/training-modules: intro-to-R-tidyverse/01-intro_to_base_R.Rmd

## Data frames

_Data frames are one of the most useful tools for data analysis in R._ 
They are tables which consist of rows and columns, much like a _spreadsheet_. 
Each column is a variable which behaves as a _vector_, and each row is an observation. 
We will begin our exploration with dataset about penguins from the [`palmerpenguins` package](https://allisonhorst.github.io/palmerpenguins/). 

> To use this dataset, we will need to extract it from the `palmerpenguins` using a `::` (more on this later).

Author: Yes. This is good thing to streamline, but probably outside the scope of this PR. I'll make an issue for it.

> To use this dataset, we will need to extract it from the `palmerpenguins` using a `::` (more on this later).

Author: #413

```{r penguin-library}
penguins <- palmerpenguins::penguins
```

> `penguins` is a data frame with measurements and information on penguins of three different species.

Reviewer: I think this sentence is redundant with line 389; we can probably delete it.

> ![](diagrams/lter_penguins.png)

Reviewer: alt text is nice to have, and she asks for credit.
```suggestion
![drawings of penguin species](diagrams/lter_penguins.png) Artwork by @allison_horst
```

> _Data frames are the most fundamental unit of data analysis in R._ 

Reviewer: I want to rephrase this; it bothered me the last time I looked at it.

```suggestion
_Data frames are one of the most useful tools for data analysis in R._ 
```

They are tables which consist of rows and columns, much like a _spreadsheet_. 
Each column is a variable which behaves as a _vector_, and each row is an observation. 

> We will begin our exploration with dataset about penguins from the `palmerpenguins` package. 

Reviewer: You will need to file a separate PR adding this to `renv.lock`

> We will begin our exploration with dataset about penguins from the `palmerpenguins` package. 

Reviewer: I think we should have a little bit of explanation of the dataset here. At least a link to https://allisonhorst.github.io/palmerpenguins/ and maybe even a picture?

![Penguins](https://allisonhorst.github.io/palmerpenguins/reference/figures/lter_penguins.png) Artwork by @allison_horst.

To use this dataset, we will need to attach the package with `library()` function (more on this later).

```{r penguin-library}

> library(palmerpenguins)

Reviewer: The trouble with doing it this way is that the data frame doesn't appear in the environment pane. I think we should probably do this with an assignment and brief explanation that we are pulling the data from a package.
```suggestion
penguins <- palmerpenguins::penguins
```

This provides summary statistics for each column:

```{r penguins-summary}
summary(penguins)
```

This provides a short view of the **str**ucture and contents of the data frame.

```{r penguins-str}
str(penguins)
```

You'll notice that the column `species` is a _factor_: This is a special type of character variable that represents distinct categories known as "levels". 
We have learned here that there are three levels in the `species` column: Adelie, Chinstrap, and Gentoo.
We might want to explore individual columns of the data frame more in-depth. 
We can examine individual columns using the dollar sign `$` to select one by name:

```{r penguins-subset}
# Extract bill_length_mm as a vector
penguins$bill_length_mm

# indexing operators can be used too
penguins$bill_depth_mm[1:10]
```

We can perform our regular vector operations on columns directly.

```{r penguins-col-mean, live = TRUE}
# calculate the mean of the bill_length_mm column

> mean(penguins$bill_length_mm, 

Author: I'm debating about leaving it like this and just gliding right through this argument OR adding some more steps to this and asking the participants whether they notice that their are `NA`s in this set.

Author: I guess main question is how much time we want to devote to this `NA` thing.

Reviewer: I think it is fine as is. This is a place where live coding should work well. 

Note that the comment you have here will be stripped out, as only full line comments are kept. In this case, I think this is a good thing! It means that somebody looking at the rendered notebook will see the full explanation, but live participants won't see it before somethin goes 'wrong'.

Author: Ah good point. I wasn't thinking about the `live`-ness part.

This provides summary statistics for each column:

```{r penguins-summary}
summary(penguins)
```

This provides a short view of the **str**ucture and contents of the data frame.

```{r penguins-str}
str(penguins)
```

You'll notice that the column `species` is a _factor_: This is a special type of character variable that represents distinct categories known as "levels". 
We have learned here that there are three levels in the `species` column: Adelie, Chinstrap, and Gentoo.
We might want to explore individual columns of the data frame more in-depth. 
We can examine individual columns using the dollar sign `$` to select one by name:

```{r penguins-subset}
# Extract bill_length_mm as a vector
penguins$bill_length_mm

# indexing operators can be used too
penguins$bill_depth_mm[1:10]
```

We can perform our regular vector operations on columns directly.

```{r penguins-col-mean, live = TRUE}
# calculate the mean of the bill_length_mm column

> mean(penguins$bill_length_mm)

Reviewer: Even though this code is being removed, we should have the final 'correct' answer in the rendered version.
```suggestion
mean(penguins$bill_length_mm,
     na.rm = TRUE) # remove missing values before calculating the mean
```

```

We can also calculate the full summary statistics for a single column directly. 

```{r penguins-col-summary, live = TRUE}
# show a summary of the bill_length_mm column
summary(penguins$bill_length_mm)
```

Extract `Species` as a vector and subset it to see a preview.

```{r penguins-col-subset, live = TRUE}
# get the first 10 values of the Species column
penguins$species[1:10]
```

And view its _levels_ with the `levels()` function.

> ```{r}

Reviewer: Missing a chunk name.
```suggestion
```{r penguin-levels}
```
